### PR TITLE
Rest of Bank 4 code disassembled.

### PR DESCRIPTION
**Pull request by 2Tie**
As mentioned in the commit, lines 899, 1538, and 1631 in engine/menu/start_menu.asm have hardcoded jumps that i couldn't get to work otherwise. There's also a few offsets and constants i wasn't sure how or where to define, and the end of the file has a lot of tile IDs that i feel could be organized better. I do plan on giving this another pass in the next day or two to clean up stray comments and rename some labels.

### CODE REVIEW

**Comment by ISSOtm** [comment]
For lines 899 and 1538, the local label you're attempting to jump to isn't in the same scope (different global labels).
Replace with `jr HandleSelectedPokemon.partypromptreturn`, requires RGBDS 0.3.7 IIRC.
Currently investigating line 1631.

**Comment by ISSOtm** [comment]
For line 1631, you made a funny mistake: the jr is located at `04:6BC3`, therefore the jump's base address is $6BC5.

`18 BF` doesn't jump to $6BC5 - $BF = $6B06 (coincidentally where `.PartySelectionRLInputs` is), but to $6BC5 - (-$BF) = $6BC5 - $41 = $6B84 = `.DrawMovePokeText`

File: engine/menu/start_menu.asm [hunk 1]
```diff
@@ -0,0 +1,2258 @@
...
+	dbw BANK(Function241ef), Function241ef
+	
+	dw KeyItemsPocketHeader
+	dw wBackpackAndKeyItemsCursor
+	dw wBackpackAndKeyItemsScrollPosition 
+	
+KeyItemsPocketHeader: ; 04:5F93
+	db MENU_BACKUP_TILES ; flags 
+	menu_coords 03, 03, $11, $0A
+	dw .keypocketdata
```
Comment (luckytyphlosion) [inline engine/menu/start_menu.asm]: no all lowercase labels, use .CamelCase

File: engine/menu/start_menu.asm [hunk 2]
```diff
@@ -0,0 +1,2258 @@
...
+	dw wRegularItemsScrollPosition 
+	
+ToolsPocketHeader: ; 04:5F75
+	db MENU_BACKUP_TILES ; flags 
+	menu_coords 03, 03, $11, $0A 
+	dw .toolspocketdata 
+	db 1 
+	
+.toolspocketdata ; 04:5F7D
+	db $AD
```
Comment (luckytyphlosion) [inline engine/menu/start_menu.asm]: lowercase hex

File: engine/dumps/bank03.asm [hunk 3]
```diff
@@ -2174,8 +2174,8 @@ Function60a0: ; 03:60a0
 	jr c, .sub_e155
 	push de
 	ld b, $00
-	ld a, BANK(Function113f4)
-	ld hl, Function113f4
+	ld a, BANK(NamingScreen)
```
Comment (luckytyphlosion) [inline engine/dumps/bank03.asm]: farcall

File: wram.asm [hunk 4]
```diff
@@ -1224,8 +1256,8 @@ wda5f:: db ; da5f
 
 SECTION "DA83", WRAM0[$DA83]
 
-wUnknownListLengthda83:: db ; da83
-wUnknownListda84:: db ; da84
+wBoxListLength:: db ; da83
+wBoxList:: ds $1E ; da84
```
Comment (luckytyphlosion) [inline wram.asm]: ds MONS_PER_BOX

File: engine/menu/text_entry.asm [hunk 5]
```diff
@@ -0,0 +1,1050 @@
...
+
+SetupMail: ; 04:5C31
+	call ClearBGPalettes
+	ld b, 8
+	call GetSGBLayout
+	call DisableLCD
+	call LoadNamingScreenGFX
+	ld de, vChars1
+	ld hl, BoldAlphabetGFX
+	lb bc, $02, $E0
```
Comment (luckytyphlosion) [inline engine/menu/text_entry.asm]: ld bc, $2e/46 tiles, or (BoldAlphabetGFXEnd - BoldAlphabetGFX)

**Comment by obskyr** [comment]
This didn't build… Apparently, `x EQU "円"`, for example, doesn't use the charmap as you might expect – instead, it sets `x` to the *Unicode* value of that character. In this case, `x` would be set to `$5186`. I'm guessing it's just taking the raw bytes between those quotation marks and interpreting them as an integer. I don't blame ya, though – that's a heck of a weird bit of behavior. All fixed up now.

**Comment by obskyr** [comment]
Ah, in fact, it looks like [that was fixed in a more recent version of RGBDS](https://github.com/rednex/rgbds/issues/293). I should upgrade.

**Comment by 2Tie** [comment]
i came across that this morning when i was starting work on bank 5 on my laptop, which was weird because i'm certain it built fine on my other computer that i did bank 4 on.

**Comment by 2Tie** [comment]
oh, that would explain it

**Comment by obskyr** [comment]
Are the strings there meaningful, by the way? Do `"♂"`, `"♀"`, and `"円"` have any connection to the context, or might it just as well just be hex values? If the former, I'll put it back in with a note in the readme to use a newer version of RGBDS.

**Comment by 2Tie** [comment]
the first two signify the graphic glyphs that get replaced with the ones the constants name (hyphen and underscore) in the naming screen, whereas the yen one is essentially just a hex value to mark the current end of the name in the prompt. for the male/female symbol i think either way would work fine, but the yen should probably stay the hex value, now that i think about it

**Comment by obskyr** [comment]
Alright, I'll put `"♂"` and `"♀"` back in, then.

**Pull request merged**